### PR TITLE
Added zero-padding to timestamp output

### LINKED ISSUE #31272: tsserver log timestamps are not properly zero-padded
**TypeScript Version:**  3.5.0-dev.20190504

**Search Terms:** tsserver log timestamp format zero padding

**Code**
The code for _nowString()_ in _/src/jsTyping/shared.ts_ seems to be the culprit:
```ts
// Existing code:
return `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}.${d.getMilliseconds()}`;

// May I suggest perhaps:
return `${d.getHours().toString().padStart(2, '0')}:${d.getMinutes().toString().padStart(2, '0')}:${d.getSeconds().toString().padStart(2, '0')}.${d.getMilliseconds().toString().padStart(3, '0')}`;
```

**Expected behavior:**
The hours/minutes/seconds/millisecond parts of the timestamps in the tsserver.log and ti-nnnn.log files should be zero-padded like this:
[17:02:03.999]
[17:02:04.000]
[17:02:04.001]

**Actual behavior:**
[17:**2**:**3**.999] 
[17:**2**:**4**.**0**]   
[17:**2**:**4**.**1**]	

**Playground Link:**  N/A

**Related Issues:** 
Issue #27428: Pretty time format does not use local time format

### PR DESCRIPTION
**Pull request by rhillefeld**
Fixes #31272
Fixed the output of `nowString()` to properly zero-pad individual timestamp components (hours, minutes, seconds, milliseconds). This affects the log output of _tsserver_.

### CODE REVIEW

**Comment by sandersn** [comment]
The code looks fine to me, but @sheetalkamat @amcasey or @uniqueiniquity will know what the ramifications of changing the format are.

**Review by amcasey — CHANGES_REQUESTED** [review]
Love the concept, but please extract a function.  Consider mimicking/unifying with [padLeft](https://github.com/microsoft/TypeScript/blob/ef1481c8a4777b3bfd242be7b717533b3ca5d149/src/compiler/core.ts#L2253).

**Comment by rhillefeld** [comment]
Thank you for your feedback. I didn't realize those functions existed already and I'm a fan of unifying similar capabilities where it makes sense. With this update, we can pad using any string, similar to padStart() and padEnd() in JS. 

Even though the original bug didn't need it, I also updated `padRight()` for consistency. Most of the code is obviously the same as `padLeft()` but I thought refactoring out just those few lines in common was overkill for this case. 

I'd like to test this further but wanted your feedback on this change before I spend too much time doing that...

File: src/compiler/core.ts [hunk 1]
```diff
@@ -2250,18 +2250,40 @@ namespace ts {
...
-    export function padLeft(s: string, length: number) {
-        while (s.length < length) {
-            s = " " + s;
+
+    /**
+     * Returns 's' left-padded with copies of 'padString' until it reaches the given 'length'.
+     * If omitted, 'padString' defaults to a single space.
+     * If 'length' is less than or equal to 's.length' or 'padString' is empty, returns 's' unchanged.
+     */
+    export function padLeft(s: string, length: number, padString = " ") {
```
Comment (amcasey) [inline src/compiler/core.ts:2259]: Personally, I would probably use a simpler, less general implementation for an internal helper like this.  For example, you could declare `padString: " " | "0" = " "` and then you wouldn't need to deal with length checks on `padString`.  In the unlikely event that someone needs a different padding character, it'll be obvious to them how to proceed.

File: src/compiler/core.ts [hunk 2]
```diff
@@ -2250,18 +2250,40 @@ namespace ts {
...
 
-    export function padRight(s: string, length: number) {
-        while (s.length < length) {
-            s = s + " ";
+    /**
+     * Returns 's' right-padded with copies of 'padString' until it reaches the given 'length'.
+     * If omitted, 'padString' defaults to a single space.
+     * If 'length' is less than or equal to 's.length' or 'padString' is empty, returns 's' unchanged.
+     */
+    export function padRight(s: string, length: number, padString = " ") {
```
Comment (amcasey) [inline src/compiler/core.ts:2277]: Thanks for keeping the helpers in sync.

**Review by amcasey — COMMENTED** [review]
Thanks for the update!  One more suggestion, but looks good.

File: src/compiler/core.ts (hunk 1 above)
Comment (rhillefeld) [reply src/compiler/core.ts:2259]: I hear what you are saying, no problem. Kinda why I wanted to get your thoughts before banging this around any more!

File: src/compiler/core.ts (hunk 2 above)
Comment (rhillefeld) [reply src/compiler/core.ts:2277]: In light of your previous comment about limiting this to `" "` or `"0"`, does it make sense to do this for `padRight()` anymore? Can't think of a use case where someone would want to pad a string with zeroes on the right... and yet, it _feels_ weird not to have the two functions match.

Comment (amcasey) [reply src/compiler/core.ts:2277]: Personally, I would have them match, but just change `" " | "0"` to `" "` in `padRight`.

**Review by amcasey — APPROVED** [review]
Nice!  Thanks for the changes.

**Comment by amcasey** [comment]
@rhillefeld Happy to merge if you're done.

**Comment by rhillefeld** [comment]
@amcasey yes, that should do it. Thanks!

**Pull request merged**